core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -247,26 +247,25 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
   @Override
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
     if (!isValidIdentifier(identifier)) {
-      throw new NoSuchTableException("Invalid identifier: %s", identifier);
+      LOG.error("Invalid identifier: %s", identifier);
+      return false;
     }
 
     Path tablePath = new Path(defaultWarehouseLocation(identifier));
     TableOperations ops = newTableOps(identifier);
-    TableMetadata lastMetadata;
-    if (purge && ops.current() != null) {
-      lastMetadata = ops.current();
-    } else {
-      lastMetadata = null;
+    TableMetadata lastMetadata = ops.current();
+    if (lastMetadata == null) {
+      LOG.error("Not an iceberg table: %s", identifier);
+      return false;
```

Comment:
I would prefer keeping the returns in the same if block as the other return statement. I tend to always lean towards single or very closely located exits from functions so it's clear that you either return this or that.

Like if you had it as
```
if (lastMetadata == null) {
  LOG.error(....)
  return false
} else {
  return fs.delete(...)
}
```

Reply:
Have tried doing something similar in the latest commit, please review.

Reply:
Looks like this is fixed to me.